SignalServiceKit/src/Util/TypingIndicators.swift

```diff
@@ -310,27 +332,32 @@ public class TypingIndicatorsImpl: NSObject, TypingIndicators {
 
     // MARK: -
 
-    // Map of (thread id and recipient id)-to-(device id)-to-IncomingIndicators.
-    private var incomingIndicatorsMap = [String: [UInt: IncomingIndicators]]()
+    // Map of (thread id)-to-(recipient id and device id)-to-IncomingIndicators.
+    private var incomingIndicatorsMap = [String: [String: IncomingIndicators]]()
```

Comment:
There's a bit of complexity here.  We want to track the typing state for each (thread, sender, device), but we want to query that in this was: "which recipient (if any) is typing in thread X?  If more than one recipient is typing, give me the one that started typing first".  

Therefore, I used a map-of-maps.

Reply:
TIOLI but typealiases might make this clearer:

```
typealias ThreadId = String
typealias ReceipientDeviceKey = String
private var incomingIndicatorsMap = [ThreadId: [ReceipientDeviceKey: IncomingIndicators]]()

```